enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/MergeEnricher.java

```diff
@@ -1,104 +0,0 @@
-/**
- * Copyright 2016 Red Hat, Inc.
- *
- * Red Hat licenses this file to you under the Apache License, version
- * 2.0 (the "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
- * implied.  See the License for the specific language governing
- * permissions and limitations under the License.
- */
-package io.fabric8.maven.enricher.standard;
-
-import io.fabric8.kubernetes.api.model.HasMetadata;
-import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
-import io.fabric8.maven.core.util.Configs;
-import io.fabric8.maven.core.util.kubernetes.KubernetesHelper;
-import io.fabric8.maven.core.util.kubernetes.KubernetesResourceUtil;
-import io.fabric8.maven.enricher.api.BaseEnricher;
-import io.fabric8.maven.enricher.api.MavenEnricherContext;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-/**
- * Merges local resources with dependent resources which have the same name.
- * <p>
- * e.g. if you wish to take a dependent set of resources and tweak them a little bit.
- *
- * (used by ipaas-platform/packages/console)
- *
- * TODO: Don't use a config enable, but use it in profiles so that its only included if required.
- * Also consider to move it to f8- since it adds some dependencies to special lifecycle flows like if that fabric8:app-catalog
- * is called. This should not be here.
- */
-public class MergeEnricher extends BaseEnricher {
-
-    public MergeEnricher(MavenEnricherContext buildContext) {
-        super(buildContext, "fmp-merge");
-    }
-
-    @Override
-    public void adapt(KubernetesListBuilder builder) {
-        List<HasMetadata> items = builder.buildItems();
-        Map<String, Map<String, HasMetadata>> kindMaps = new HashMap<>();
-
-        List<HasMetadata> removeList = new ArrayList<>();
-        for (HasMetadata item : items) {
-            String kind = KubernetesHelper.getKind(item);
-            String name = KubernetesHelper.getName(item);
-            Map<String, HasMetadata> map = kindMaps.get(kind);
-            if (map == null) {
-                map = new HashMap<>();
-                kindMaps.put(kind, map);
-            }
-            HasMetadata old = map.get(name);
-            if (old != null) {
-                HasMetadata removeItem = mergeEntities(old, item);
-                if (removeItem != null) {
-                    removeList.add(removeItem);
```

Comment:
If two items have been merged successfully (return value != null) then it shoud be removed from the list ? Doesn't make sense at all...